sdk/metricsadvisor/Azure.AI.MetricsAdvisor/src/Models/DataFeed/DataFeed.cs

```diff
@@ -53,6 +53,36 @@ internal DataFeed(DataFeedDetail dataFeedDetail)
             Viewers = dataFeedDetail.Viewers;
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DataFeed"/> class for mocking
+        /// purposes.
+        /// </summary>
+        internal DataFeed(string id, DataFeedStatus? status, DateTimeOffset? createdOn, string creator, bool? isAdministrator,
```

Comment:
is there any advantage of adding a new constructor with individual properties instead of creating a `DataFeedDetail` object in the `MAModelFactory` class?

Reply:
We can use the new constructor to directly set the `MetricIds` property, which is just a convenient dictionary calculated by us from other properties. We expect users to use it heavily. This is not possible via `DataFeedDetail`.